docs/contributing/component-js-configuration-options.md

```diff
@@ -0,0 +1,192 @@
+# Adding JavaScript configuration options to components
+
+Configuration options permit developers to pass additional information into a component's JavaScript, allowing them to change how the component might work or appear without needing to create their own fork.
+
+Configuration options straddle the line between Nunjucks, HTML and JavaScript.
+
+## Preparing the component's JavaScript
+
+First, ensure the entry function for component has a parameter for passing in a configuration object.
+
+```javascript
+function Accordion ($module, config) {
+  ...
+}
+```
+
+Within the entry function, you'll then create an object with the default configuration for the component. These are what options will be used if no other options have been provided.
+
+```javascript
+var defaultConfig = {
+  rememberExpanded: true
+}
+```
+
+Next, use the `mergeConfigs` helper to combine the default config and the configuration object. This is assigned globally (using `this`) so that it's accessible anywhere within the component's JavaScript.
```

Comment:
Would it work to structure this doc with numbered steps in each section?

Reply:
Where it says 'This is assigned globally...' is it referring to the helper, the default config, the configuration object or the result of the merging done by the helper?

Reply:
> Would it work to structure this doc with numbered steps in each section?

I considered this, but as some steps contain multiple paragraphs of information and code blocks, it isn't easy to semantically mark it up as a numbered list, so I didn't. 

> Where it says 'This is assigned globally...' is it referring to the helper, the default config, the configuration object or the result of the merging done by the helper?

The result of the merging done by the helper.

Reply:
Yeah, as  worked my way down I could see that numbering would only work in some places making the whole doc rather inconsistent.

> The result of the merging done by the helper.

Thanks